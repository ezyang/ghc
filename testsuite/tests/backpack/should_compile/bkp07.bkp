package p where
    signature A where
        foo :: a -> a

package q where
    include p
    module B where
        import A
        bar x = foo (x + x)
