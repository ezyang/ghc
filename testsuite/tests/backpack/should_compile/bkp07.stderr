[1 of 2] Processing p
/--- Shape for p
provides:
requires: A -> foo
\---
  [1 of 1] Compiling A[sig]           ( p/A.hsig, nothing )
[2 of 2] Processing q
/--- Shape for q
provides: B -> main-q(hole:A):B
               bar
requires: A -> foo
\---
  [1 of 2] Compiling A[sig]           ( q/A.hsig, nothing )
  [2 of 2] Compiling B                ( q/B.hs, nothing )
